### DISCUSSION TITLE
Ability to target children elements via hooks?

### DISCUSSION THREAD

**Discussion by mryechkin** [css-hooks/css-hooks#99] [Ideas]
This is more of a question, and is perhaps completely out of scope for CSS Hooks, but nonetheless I'm curious if something like this is doable - or if you have suggestions on how to enable this.

Let's say I have a `Button` component for my design system that has the following (simplified) structure:

```jsx
const Button = ({ children, icon, ...props }) => (
  <button className="acme-Button" {...props}>
    {icon && <div className="acme-Button-icon">{icon}</div>}
    {children}
  </button>
);
```

`acme-Button` and `acme-Button-icon` have some default styles defined in a CSS bundle (ie. the design system defaults shipped to the user).

Now, if I'd like my users to override the icon color, they can do so via CSS:

```css
.acme-Button-icon {
  color: hotpink;
}
```

And if they're using Tailwind CSS, then they can do something like this (using the arbitrary selector feature):

```jsx
import { Button } from 'acme-ds';
import { Icon } from 'acme-icons';

// ...

<Button className="[&_.acme-Button-icon]:text-[hotpink]" icon={<Icon />}>
  Click Me
</Button>
```

Would I be able to do something similar with CSS Hooks? Something like this:

```jsx
<Button icon={<Icon />} style={css({ '& .acme-Button-icon': { color: 'hotpink' } })}>
  Click Me
</Button>
```

I tried defining `"& .acme-Button-icon"` as a custom hook, but that doesn't seem to have any effect. So I'm just wondering if this is even doable, or completely outside of the scope? Is there anything I can do in userland to enable something like this?

Appreciate any insight!

**Comment by robinweser** [comment]
Unfortunately, this is not possible due to inline style limitations.
CSS hooks compile to inline styles after all, which are self-centric and cannot target any other elements.
You can conditionally style the icon based on its parent if that helps, but all styles always apply directly to the styled element.

In regards to the icon styling you can do sth. like:
```js
const style = {
  // default style
  ".acme-Button &": {
    color: "hotpink"
  }
}
```
but you'd always have to apply it to the icon, not the button.

@nsaunders I think we should clarify that in the docs as well. The self-centric nature of hooks or however you'd call that.

**Reply by mryechkin** [reply]
Makes sense, thanks for confirming. I had pretty much expected as much, but thought I would check. Appreciate it!

**Comment by nsaunders** [comment]
Thanks for the question @mryechkin and the response @robinweser.

Actually @robinweser based on your suggestion, I already started a FAQ page - see [Why can't hooks affect descendant elements?](https://css-hooks.com/docs/faq/#why-can39t-hooks-affect-descendant-elements). Please feel free to suggest any edits or additional FAQ items you'd like to address. :-)

**Reply by mryechkin** [reply]
Sorry, I definitely missed that FAQ 🤦‍♂️ That would've answered my question! Haha. Thanks

**Reply by nsaunders** [reply]
Don't worry! It's a recent addition and a WIP as you can see. 🙂

**Reply by robinweser** [reply]
Oh, I actually also didn't check the latest changes 😅 even better then!
